tensorflow/core/kernels/argmax_op.cc

```diff
@@ -56,7 +56,7 @@ class ArgOp : public OpKernel {
                     "dim must be a scalar, but received tensor of shape: ",
                     dimension.shape().DebugString()));
 
-    const int32_t dim = internal::SubtleMustCopy(dimension.scalar<int32>()());
+    const Tidx dim = internal::SubtleMustCopy(dimension.scalar<Tidx>()());
     const int input_dims = input.dims();
 
     int axis = dim < 0 ? dim + input_dims : dim;
```

Comment:
As Mihai mentioned would the problem not arise here?

Reply:
There seems to be one more problematic code here.
https://github.com/tensorflow/tensorflow/blob/50bfa12ba07bcd797be1d3841d059b4418a565de/tensorflow/core/ops/math_ops.cc#L1125-L1130

My assumption is when you call a tensor value like this `tensor.scalar<Ttype>` we need to check `Ttype` should be same as the `dtype` of `tensor`. For example refer above snippet. 

It seems this constraint if while extracting values from tensor.

Reply:
Hi @sagunb Can you please take a look at this PR? Thank you!

Reply:
It's not @sagunb who needs to look at the PR but @SuryanarayanaY (the author)

Reply:
Hi @sagunb Can you please take a look at this PR? Thank you!

Reply:
Hi @sagunb Can you please take a look at this PR? Thank you!